<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>830</width>
    <height>520</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>830</width>
    <height>520</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>830</width>
    <height>520</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:  rgb(112, 144, 124);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>820</width>
     <height>510</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>820</width>
     <height>510</height>
    </size>
   </property>
   <widget class="s21::MazeWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>500</width>
      <height>500</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>500</width>
      <height>500</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>500</width>
      <height>500</height>
     </size>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>520</x>
      <y>10</y>
      <width>300</width>
      <height>500</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>300</width>
      <height>500</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>300</width>
      <height>500</height>
     </size>
    </property>
    <property name="toolTip">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="whatsThis">
     <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
    <property name="styleSheet">
     <string notr="true">QSlider::groove:horizontal {
    border: 1px solid #999999;
    height: 8px;
    background: qlineargradient(x1:0, y1:0, x2:0, y2:1, stop:0 rgb(223, 238, 234), stop:1 #c4c4c4);
    margin: 2px 0;
}

QSlider::handle:horizontal {
    background: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 rgb(247, 213, 79), stop:1 #8f8f8f);
    border: 1px solid #5c5c5c;
    width: 18px;
    margin: -2px 0;
    border-radius: 3px;
}

QTabBar::tab {
color: rgb(0, 0, 0);
background: rgb(113, 144, 124); 
border: 5px;
border-color: rgb(0, 0, 0);
padding: 5px;
}

QTabBar::tab:selected { 
  background: rgb(223, 238, 234); 
}

QSpinBox {
color: rgb(0, 0, 0);
background-color:  rgb(223, 238, 234);
}

QLabel {
color: rgb(0, 0, 0);
}

QWidget {
background-color: rgb(179, 209, 201);
}

QPushButton {
color: rgb(0, 0, 0);
background-color: qlineargradient(x1:0, y1:0, x2:1, y2:1, stop:0 rgb(247, 213, 79), stop:1 rgb(226, 160, 57));
border-color: rgb(0, 0, 0);
 border-style: solid;
 border-width:1px;
 border-radius:10px;
}

QPushButton:hover {
color: rgb(0, 0, 0);
background-color: rgb(241, 188, 70);
border-color: rgb(0, 0, 0);
border-style: solid;
border-width:1px;
border-radius:10px;
}

QPushButton:pressed {
 color: rgb(0, 0, 0);
background-color: rgb(226, 160, 57);
border-color: rgb(0, 0, 0);
 border-style: solid;
 border-width:1px;
 border-radius:10px;
}</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="maze">
     <attribute name="title">
      <string>Maze</string>
     </attribute>
     <widget class="QPushButton" name="pushButtonGenerate">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>370</y>
        <width>250</width>
        <height>80</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>250</width>
        <height>80</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>250</width>
        <height>80</height>
       </size>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButtonDeletePath">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>145</y>
        <width>100</width>
        <height>45</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Delete path</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_lenght_maze">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>40</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_height_maze">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>40</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Height</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="OpenFileMaze">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>15</y>
        <width>125</width>
        <height>20</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Open File</string>
      </property>
     </widget>
     <widget class="QPushButton" name="SaveFIleMaze">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>15</y>
        <width>125</width>
        <height>20</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>Save file</string>
      </property>
     </widget>
     <widget class="QSlider" name="lenght_maze_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>60</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="pageStep">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="lenght_maze_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>60</y>
        <width>25</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="QSlider" name="height_maze_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>110</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="pageStep">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="height_maze_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>110</y>
        <width>25</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="cave">
     <attribute name="title">
      <string>Cave</string>
     </attribute>
     <widget class="QPushButton" name="GenerateCave">
      <property name="geometry">
       <rect>
        <x>195</x>
        <y>355</y>
        <width>80</width>
        <height>40</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QLabel" name="lebel_steps_delay">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>405</y>
        <width>80</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Steps delay</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="milliseconds_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>425</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="singleStep">
       <number>20</number>
      </property>
      <property name="pageStep">
       <number>1000</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="milliseconds_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>425</y>
        <width>35</width>
        <height>21</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="maximum">
       <number>1000</number>
      </property>
      <property name="singleStep">
       <number>20</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
     <widget class="QSlider" name="limit_birth_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>160</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="pageStep">
       <number>7</number>
      </property>
      <property name="value">
       <number>4</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_lenght_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>40</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Length</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_height_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>40</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Height</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="lenght_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>60</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>50</width>
        <height>20</height>
       </size>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="pageStep">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="lenght_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>60</y>
        <width>25</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="QSlider" name="height_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>110</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="pageStep">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="height_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>110</y>
        <width>25</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>10</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_birth_limit_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>140</y>
        <width>60</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Birth limit</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSpinBox" name="limit_birth_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>160</y>
        <width>25</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="value">
       <number>4</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_limit_of_death_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>190</y>
        <width>90</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Limit of death</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="limit_of_death_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>210</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="pageStep">
       <number>7</number>
      </property>
      <property name="value">
       <number>4</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="limit_of_death_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>210</y>
        <width>25</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
      <property name="singleStep">
       <number>1</number>
      </property>
      <property name="value">
       <number>4</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_initialization_chance_cave">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>240</y>
        <width>130</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>Initialization chance</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="initialization_chance_Slider">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>260</y>
        <width>235</width>
        <height>25</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="pageStep">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QSpinBox" name="initialization_chance_spinBox">
      <property name="geometry">
       <rect>
        <x>255</x>
        <y>260</y>
        <width>25</width>
        <height>20</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="contextMenuPolicy">
       <enum>Qt::NoContextMenu</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color:  rgb(223, 238, 234);</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QPushButton" name="AutoStepCave">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>355</y>
        <width>80</width>
        <height>40</height>
       </rect>
      </property>
      <property name="text">
       <string>Auto</string>
      </property>
     </widget>
     <widget class="QPushButton" name="OneStepCave">
      <property name="geometry">
       <rect>
        <x>105</x>
        <y>355</y>
        <width>80</width>
        <height>40</height>
       </rect>
      </property>
      <property name="text">
       <string>Step</string>
      </property>
     </widget>
     <widget class="QPushButton" name="OpenFileCave">
      <property name="geometry">
       <rect>
        <x>15</x>
        <y>15</y>
        <width>125</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Open File</string>
      </property>
     </widget>
     <widget class="QPushButton" name="SaveFileCave">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>15</y>
        <width>125</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Save file</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::MazeWidget</class>
   <extends>QWidget</extends>
   <header location="global">view/mazewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
